packages/create-sitecore-jss/src/templates/angular-xmcloud/scripts/config/plugins/computed.ts

```diff
@@ -0,0 +1,43 @@
+import { JssConfig } from 'lib/config';
+import { ConfigPlugin } from '..';
+import { constantCase } from 'constant-case';
+import { constants } from '@sitecore-jss/sitecore-jss-angular/cjs';
+
+/**
+ * This plugin will set computed config props.
+ * The "graphQLEndpoint" is an example of making a _computed_ config setting
+ * based on other config settings.
+ */
+class ComputedPlugin implements ConfigPlugin {
+  // should come after other plugins (but before fallback)
+  order = 10;
+
+  async exec(config: JssConfig) {
+    const proxyBuildPath = process.env[`${constantCase('proxyBuildPath')}`]?.replace(/\/$/, '');
+    const proxyHost = process.env[`${constantCase('proxyHost')}`];
+
+    const production = process.env.JSS_MODE === constants.JSS_MODE.PRODUCTION;
+
+    let graphQLEndpoint;
+    let computed = `${config.graphQLEndpoint || config.sitecoreApiHost}${
+      config.graphQLEndpointPath
+    }`;
+
+    if (production) {
+      // When build for production (using `jss build`)
+      // The client side requests will be proxied through the node-xmcloud-proxy and passed to the Sitecore instance
+      // The server side requests will be executed against the Sitecore instance directly
+      graphQLEndpoint = `\${typeof window === 'undefined' ? '${computed}' : '${proxyHost}${config.graphQLEndpointPath}'}`;
```

Comment:
Actually ... when in production, don't we want _both_ the server and client side requests going through the proxy? This is how we are eventually going to perform personalization (which needs to happen on both sides).

Reply:
I think I'm just not clear in which case we would be using a server request here. If we are always running through the SSR proxy, seems like never (it will always be the client side case).
 
And if so, could we simplify this to assume client-side. e.g.
```
graphQLEndpoint = `${proxyHost}${config.graphQLEndpointPath}`;
```

Reply:
@ambrauer The problem here, that proxy uses layoutServiceFactory and dictionaryServiceFactory, that are using this config. So this is the case when we are on the server and request should go directly to CM

Reply:
@ambrauer I'm thinking if it's a reason to move this logic in clientFactory in this case?

Reply:
Moved to graphql-client-factory